File: core/src/test/java/org/vizzini/core/InputValidatorTest.java
package org.vizzini.core;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;
import static org.junit.Assert.fail;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.junit.Test;

/**
 * Provides tests for the <code>InputValidator</code> class.
 */
public class InputValidatorTest
{
    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyArray()
    {
        // Setup.
        final String[] array = new String[2];
        array[0] = "one";
        array[1] = "two";

        // Run / Verify.
        InputValidator.validateNotEmpty("array", array);
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyArrayEmpty()
    {
        // Setup.
        final String[] array = new String[0];

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("array", array);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("array is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyArrayNull()
    {
        // Setup.
        final String[] array = null;

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("array", array);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("array is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyList()
    {
        // Setup.
        final List<String> list = new ArrayList<String>();
        list.add("one");

        // Run / Verify.
        InputValidator.validateNotEmpty("list", list);
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyListEmpty()
    {
        // Setup.
        final List<String> list = new ArrayList<String>();

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("list", list);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("list is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyListNull()
    {
        // Setup.
        final List<String> list = null;

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("list", list);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("list is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyMap()
    {
        // Setup.
        final Map<String, String> map = new HashMap<String, String>();
        map.put("one", "two");

        // Run / Verify.
        InputValidator.validateNotEmpty("map", map);
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyMapEmpty()
    {
        // Setup.
        final Map<String, String> map = new HashMap<String, String>();

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("map", map);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("map is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyMapNull()
    {
        // Setup.
        final Map<String, String> map = null;

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("map", map);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("map is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyString()
    {
        // Setup.
        final String string = "one";

        // Run / Verify.
        InputValidator.validateNotEmpty("string", string);
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyStringEmpty()
    {
        // Setup.
        final String string = "";

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("string", string);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("string is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotEmpty()</code> method.
     */
    @Test
    public void validateNotEmptyStringNull()
    {
        // Setup.
        final String string = null;

        // Run / Verify.
        try
        {
            InputValidator.validateNotEmpty("string", string);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("string is null or empty."));
        }
    }

    /**
     * Test the <code>validateNotNegative()</code> method.
     */
    @Test
    public void validateNotNegativeMinusOne()
    {
        // Setup.
        final int value = -1;

        // Run / Verify.
        try
        {
            InputValidator.validateNotNegative("value", value);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("value is less than zero."));
        }
    }

    /**
     * Test the <code>validateNotNegative()</code> method.
     */
    @Test
    public void validateNotNegativeZero()
    {
        // Setup.
        final int value = 0;

        // Run / Verify.
        InputValidator.validateNotNegative("value", value);
    }

    /**
     * Test the <code>validateNotNull()</code> method.
     */
    @Test
    public void validateNotNullString()
    {
        // Setup.
        final String string = "";

        // Run / Verify.
        InputValidator.validateNotNull("string", string);
    }

    /**
     * Test the <code>validateNotNull()</code> method.
     */
    @Test
    public void validateNotNullStringNull()
    {
        // Setup.
        final String string = null;

        // Run / Verify.
        try
        {
            InputValidator.validateNotNull("string", string);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("string is null."));
        }
    }

    /**
     * Test the <code>validatePositive()</code> method.
     */
    @Test
    public void validatePositiveOne()
    {
        // Setup.
        final int value = 1;

        // Run / Verify.
        InputValidator.validatePositive("value", value);
    }

    /**
     * Test the <code>validatePositive()</code> method.
     */
    @Test
    public void validatePositiveZero()
    {
        // Setup.
        final int value = 0;

        // Run / Verify.
        try
        {
            InputValidator.validatePositive("value", value);
            fail("Should have thrown an exception");
        }
        catch (final IllegalArgumentException e)
        {
            assertThat(e.getMessage(), is("value is zero or less."));
        }
    }
}

File: coreweb/src/test/js/util/InputValidatorTest.js
"use strict";
QUnit.module("InputValidator");

var LOGGER = new Logger();
var SHOULD_THROW = "Should have thrown an exception.";

QUnit.test("isArray()", function(assert)
{
    var array = [1, 2, 3, 4];
    assert.ok(!InputValidator.isArray());
    assert.ok(!InputValidator.isArray(undefined));
    assert.ok(!InputValidator.isArray(null));
    assert.ok(!InputValidator.isArray(false));
    assert.ok(!InputValidator.isArray(""));
    assert.ok(!InputValidator.isArray(12));
    assert.ok(InputValidator.isArray([]));
    assert.ok(InputValidator.isArray(array));
});

QUnit.test("isNumber()", function(assert)
{
    var array = [1, 2, 3, 4];
    assert.ok(!InputValidator.isNumber());
    assert.ok(!InputValidator.isNumber(undefined));
    assert.ok(!InputValidator.isNumber(null));
    assert.ok(!InputValidator.isNumber(false));
    assert.ok(!InputValidator.isNumber(""));
    assert.ok(InputValidator.isNumber(12));
    assert.ok(!InputValidator.isNumber([]));
    assert.ok(!InputValidator.isNumber(array));
});

QUnit.test("validateNotEmpty()", function(assert)
{
    try
    {
        InputValidator.validateNotEmpty();
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "undefined" + InputValidator.EMPTY);
    }
    try
    {
        InputValidator.validateNotEmpty("value");
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.EMPTY);
    }
    try
    {
        InputValidator.validateNotEmpty("value", undefined);
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.EMPTY);
    }
    try
    {
        InputValidator.validateNotEmpty("value", null);
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.EMPTY);
    }
});

QUnit.test("validateNotEmpty() boolean", function(assert)
{
    InputValidator.validateNotEmpty("value", false);
    InputValidator.validateNotEmpty("value", true);
    assert.ok(true);
});

QUnit.test("validateNotEmpty() string", function(assert)
{
    try
    {
        InputValidator.validateNotEmpty("value", "");
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.EMPTY);
    }

    InputValidator.validateNotEmpty("value", "string");
});

QUnit.test("validateNotEmpty() number", function(assert)
{
    try
    {
        InputValidator.validateNotEmpty("value", NaN);
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.EMPTY);
    }

    InputValidator.validateNotEmpty("value", 0);
    InputValidator.validateNotEmpty("value", 12);
});

QUnit.test("validateNotEmpty() array", function(assert)
{
    try
    {
        InputValidator.validateNotEmpty("value", []);
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.EMPTY);
    }

    InputValidator.validateNotEmpty("value", [1, 2, 3, 4]);
});

QUnit.test("validateNotEmpty() object", function(assert)
{
    var object = {};
    InputValidator.validateNotEmpty("value", object);
    assert.ok(true);
});

QUnit.test("validateNotNull() undefined", function(assert)
{
    try
    {
        InputValidator.validateNotNull();
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "undefined" + InputValidator.UNDEFINED);
    }
    try
    {
        InputValidator.validateNotNull("value");
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.UNDEFINED);
    }
    try
    {
        InputValidator.validateNotNull("value", undefined);
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.UNDEFINED);
    }
});

QUnit.test("validateNotNull() null", function(assert)
{
    try
    {
        InputValidator.validateNotNull("value", null);
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.NULL);
    }
    try
    {
        InputValidator.validateNotNull("value", null);
        throw SHOULD_THROW;
    }
    catch (e)
    {
        assert.equal(e, "value" + InputValidator.NULL);
    }
});
